src/player.cpp

```diff
@@ -10884,42 +10884,45 @@ bool player::has_enough_charges( const item &it, bool show_msg ) const
     return true;
 }
 
-bool player::consume_charges(item *used, long charges_used)
+bool player::consume_charges( item *used, long qty )
 {
-    // Non-tools can use charges too - when they're comestibles
-    const auto tool = dynamic_cast<const it_tool*>(used->type);
-    const auto comest = dynamic_cast<const it_comest*>(used->type);
-    if( charges_used <= 0 || (tool == nullptr && comest == nullptr) ) {
+    if( !has_item( used ) ) {
+        debugmsg( "Tried to consume charges for an item not in players possession" );
         return false;
     }
 
-    if( tool != nullptr && tool->charges_per_use <= 0 ) {
-        // An item that doesn't normally expend charges is destroyed instead.
-        /* We can't be certain the item is still in the same position,
-         * as other items may have been consumed as well, so remove
-         * the item directly instead of by its position. */
-        i_rem( used );
-        return true;
+    if( qty <= 0 ) {
+        debugmsg( "Tried to consume zero or negagtive charges" );
```

Comment:
Typo - "negagtive"
Pretty sure consuming 0 charges is allowed. For example, when canceling usage of an item or using an item that doesn't consume charges.

Not sure if negative charges are used anywhere.